ext/ruby_newt/ruby_newt.c

```diff
@@ -1071,13 +1159,15 @@ static VALUE rb_ext_Form_SetWidth(VALUE self, VALUE width)
   return Qnil;
 }
 
-static VALUE rb_ext_Run_Form(VALUE self)
+static VALUE rb_ext_Form_Run(VALUE self)
 {
-  newtComponent form, co;
+  newtComponent form;
+  struct newtExitStruct *es;
 
   Data_Get_Struct(self, struct newtComponent_struct, form);
-  co = newtRunForm(form);
-  return Data_Wrap_Struct(cWidget, 0, 0, co);
+  es = ALLOC(struct newtExitStruct);
+  newtFormRun(form, es);
+  return Data_Wrap_Struct(cExitStruct, 0, 0, es);
```

Comment:
You're right. For some reason I was under the impression that ruby called xfree() internally when no free function is supplied. It doesn't look like that's the case.